src/core/io_queue.cc

```diff
@@ -858,8 +858,8 @@ stream_id io_queue::request_stream(io_direction_and_length dnl) const noexcept {
 
 double internal::request_tokens(io_direction_and_length dnl, const io_queue::config& cfg) noexcept {
     struct {
-        unsigned weight;
-        unsigned size;
+        double weight;
+        double size;
```

Comment:
Floating point math is not welcome here

Reply:
well, I know, but we should welcome it here, I don't see it breaking anything here.
In this particular function, you cast to double anyway in the return stmt.
What this change addresses is the fact that for bigger configured iops and bandwidth, `disk_req_write_to_read_multiplier` and `disk_blocks_write_to_read_multiplier` end up < 1, so casting them to unsigned would make them 0, so in the end costs end up 0.
They should end up 0 anyway the closer you get to uint64_t max, but not that soon.

I think this should be split into a different PR as it addresses a different issue now that it's settled the hang needs to be fixed by fixing the loop, what do you think?